doc/compiler/control/CompilationControl.md

```diff
@@ -0,0 +1,354 @@
+is used to describe the body (e.g. start and end address, optimization
+level, etc.). Space in the Code Repository and Data Caches have to be
+allocated on an as-needed basis. Additionally, these allocations have
+to be synchronized since multiple Compilation Threads can be active
+simultaneously. As such, each Compilation Thread, as part of the
+compilation, will reserve a Code and Data Cache at the beginning of the
+codegen phase to ensure that no other thread can write to it.
+
+
+### Trampolines
+
+A trampoline is essentially a snippet of code that is used to branch to
+a location far from the current Instruction Pointer. Each Code Cache
+has a section for storing Trampolines so that methods in a
+Code Cache can share Trampolines. The Code section grows towards
+increasing addresses, while the Trampoline section grows towards
+decreasing addresses.
+
+The Trampoline section is not reserved if the JVM can guarantee that far
+jumps will never be needed. The Code Repository is fundamental to
+providing this guarantee by ensuring that Code Caches are close together
+and obviate the need for far jumps. However, platforms such as POWER
+where the maximum distance of the branch target is much shorter than
+platforms such as x86 may still need to use Trampolines.
+
+### Code Cache Reclamation
+
+When a method is recompiled, the old body eventually stops being used,
+which is a waste of space in the Code Cache. In order to maximize the
+usable space in the Code Cache, the JIT reclaims stale bodies; see the
+[Code Cache Reclamation doc](../runtime/CodeCacheReclamation.md) for
+more details.
+
+
+# Influencing Factors
+
+There are many factors that influence the decisions the Control
+Infrastructure makes. The JIT has to improve application
+performance without taking away too many resources from the
+application. The following are some of the considerations that have
+the biggest impact and influence.
+
+## JIT State
+
+The JIT maintains several states that reflect the various execution
+characteristics of an application. The Compilation Control heuristics
+depend on the state, or phase, the JIT is in. The JIT transitions
+between the following states:
+
+- IDLE_STATE
+  - Decisions focus on minimizing Compilation Thread CPU usage.
+- STARTUP_STATE
+  - Decisions focus on making Startup as fast as possible.
+- RAMPUP_STATE
+  - Decisions focus on making Rampup as fast as possible.
+- STEADY_STATE
+  - Decisions focus on maintaining throughput.
+- DEEPSTEADY_STATE
+  - Decisions focus on maintaining throughput while minimizing CPU
+    consumed by the JIT (including the Sampler Thread).
+
+Startup refers to the period of time between the start of the JVM and
+the point at which the Java Application is ready for load. Rampup
+refers to the period of time between the end of Startup and the point
+at which the Java Application is running at peak throughput. When the
+application runs at peak throughput, it is said to be in Steady State.
```

Comment:
I feel each of these sentences should move up to their corresponding state description.